compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/NonTerminalNode.java

```diff
@@ -85,12 +86,13 @@ protected int getChildPosition(int bucket) {
 
     // Create a Token for the TextChange token in the given bucket
     protected Token createToken(int bucket) {
-        STNode internaltoken = this.node.childInBucket(bucket);
-        if (internaltoken == null) {
-            return null;
+        STNode internalToken = this.node.childInBucket(bucket);
+        if (internalToken == null || internalToken.kind == SyntaxKind.NONE) {
+            // TODO Verify this change
+            return new EmptyToken(internalToken, 0, this);
```

Comment:
Ideally we would not need this `internalToken == null` at all. If something is coming as null, then thats probably a bug in the parser.